src/model/model.js

```diff
@@ -472,6 +474,49 @@ export default class Model {
 		return false;
 	}
 
+	/**
+	 * Checks whether the given {@link module:engine/model/range~Range range} or
+	 * {@link module:engine/model/element~Element element} is considered empty.
+	 *
+	 * The range or element is considered non-empty if it contains any:
+	 * 		* EmptyElement
+	 * 		* Text node containing at least one non-whitepsace character
+	 * 		* Non-plain `ContainerElement` (for example widget)
+	 * 		* Non-plain `AttributeElement` (for example comment)
+	 *
+	 * 	This method should be used to check if the element/range/editor contains any printable/meaningful content.
+	 * 	It is also the correct method to check if editor is empty.
+	 *
+	 * @param {module:engine/model/range~Range|module:engine/model/element~Element} rangeOrElement Range or element to check.
+	 * @returns {Boolean}
+	 */
+	isEmpty( rangeOrElement ) {
```

Comment:
The relation between `isEmpty()` and `hasContent()` needs to be clarified. Both these methods are defined on `Model` but they have different semantics. I wonder if we shouldn't rename `hasContent()` or `isEmpty()` to somehow differentiate between these two.

cc @pjasiun

Reply:
Let's merge `isEmpty()` into `hasContent()` (i.e. we will have a single method for that called `hasContent()`).

Right now there are 2 differences between them – check for markers and `/\S+/`. The former can be done automatically by `hasContent()`. The latter could be enabled by an `option.trimWhitespaces`.

Reply:
👍